### ISSUE TITLE
Excel 365: NVDA does not report when user is at the edge of a sheet

### ISSUE THREAD

**Issue by bhavyashah** [nvaccess/nvda#13025]
### Steps to reproduce:
1 Open a new or existing spreadsheet in Excel.
2 Navigate to a cell whose alphabet index is a and press left arrow or whose number index is 1 and press up arrow.

### Actual behavior:
Depending on the contents of the cell and cell coordinate reporting in Document Formatting Settings, NVDA will merely re-report the edge cell. The instance when this is especially inadequate is when the edge cell is empty (or has non-unique contents) and cell coordinate reporting is turned off (usually the case for me), because it is unclear if I am continuing to land on new but identical cells or if NVDA is repeating the same cell over and over again.

### Expected behavior:
If this were a table in Word or on the web, NVDA would say "edge of table." If this were Google Sheets, NVDA would say "no more cells." Here too, NVDA should behave similarly.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
NVDA 2021.3 Beta 1

#### Windows version:
Windows 10

#### Name and version of other software in use when reproducing the issue:
Excel for Microsoft 365

#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by bhavyashah** [comment]
This issue is reproducible on Excel 365 (latest updates installed) and NVDA 2023.2.
